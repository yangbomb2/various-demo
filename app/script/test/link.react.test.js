/*
  Test React Component
 */

import React from 'react';
import Link from '../components/link';
import Dummy from '../components/dummy';
import renderer from 'react-test-renderer';

test('react link changes state on hover', () => {

  const link = renderer.create(
    <Link url="http://www.facebook.com">Facebook</Link>
  );

  let tree = link.toJSON();
  expect(tree).toMatchSnapshot();

  //hover testing
  tree.props.onMouseEnter();

  tree = link.toJSON();
  expect(tree).toMatchSnapshot();

  //leavec testing
  tree.props.onMouseLeave();
  tree = link.toJSON();
  expect(tree).toMatchSnapshot();

});
