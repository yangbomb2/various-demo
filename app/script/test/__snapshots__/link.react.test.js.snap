exports[`test react link changes state on hover 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Facebook
</a>
`;

exports[`test react link changes state on hover 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Facebook
</a>
`;

exports[`test react link changes state on hover 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Facebook
</a>
`;
